transcom/mymove
# MB-7425 Bugfix -  Omit empty customer contact fields for DDASIT, DDDSIT

## Description

* This PR removes `required` from the following fields on `MTOServiceItemDestSIT`:
  * `firstAvailableDeliveryDate1`
  * `firstAvailableDeliveryDate2`
  * `timeMilitary1`
  * `timeMilitary2`

* This allows these fields to be omitted in the response for DDASIT and DDDSIT service items, instead of being returned as null or an empty string.

* The fields **are still required for the DDFSIT**, however. I added server-side validation to check for these fields.

## Reviewer Notes

For now, the validation is placed alongside similar validation for service items in `payload_to_model`. @sandy-wright and I discussed what a future refactor might look like, implementing a new validator pattern she's working on now. In the meantime, this was the simplest way to add the validation and follow an existing pattern.

## Setup

### Automated Test
* `go test ./pkg/handlers/primeapi/ --testify.m "TestCreateMTOServiceItemDestSITHandler" -v`

### Manual Test
* To test, use Postman to hit `POST /mto-service-items` and create a new DDFSIT
* Try creating a DDFSIT without one (or all) of the customer contact fields (listed above). Confirm that you get the appropriate validation error returned.
* Try creating a DDFSIT with the correct data, and confirm you can still create one.
* After successfully creating a DDFSIT, confirm that no empty or null fields are returned for DDASIT, DDDSIT.
Sample payload for DDFSIT:

```
{
    "modelType": "MTOServiceItemDestSIT",
    "moveTaskOrderID": "fb6f3259-926b-432b-a432-a995bd162e7a",
    "mtoShipmentID": "c1efb9f2-efbf-44a7-9d1f-ced7aa1551c3",
    "firstAvailableDeliveryDate1": "2021-02-02",
    "firstAvailableDeliveryDate2": "2021-05-02",
    "timeMilitary1": "0400Z",
    "timeMilitary2": "0500Z",
    "reServiceCode": "DDFSIT",
    "sitEntryDate": "2021-06-01"   
}
```

## Code Review Verification Steps

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7425) for this change
